python/smashbox/config.py

```diff
@@ -0,0 +1,99 @@
+import logging
+import os.path
+import pickle
+from pathlib import Path
+from typing import Any, Literal
+
+import yaml
+from pydantic import BaseSettings
+
+
+logger = None
```

Comment:
Global variables are the root of evil. Generally, you should try to avoid them.

With Python's logging facilities, you do not need to "cache" logger objects anyway. Python handles this internally.